game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/RemoveUnprotectedUnits.java

```diff
@@ -51,20 +51,27 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
   }
 
   public void removeUnprotectedUnits(final IDelegateBridge bridge, final BattleState.Side side) {
-    if (Properties.getTransportCasualtiesRestricted(battleState.getGameData())) {
-      checkUndefendedTransports(bridge, side);
-      checkUnprotectedUnits(bridge, side);
+    if (!Properties.getTransportCasualtiesRestricted(battleState.getGameData())) {
+      return;
     }
+    // if we are the attacker, we can retreat instead of dying
+    if (attackerHasRetreat(side)) {
+      return;
+    }
+    checkUndefendedTransports(bridge, side);
```

Comment:
Having the early return in this method is much better :+1: 

Early return is a pattern I like, I do prefer:
```
void method()  {
  if(something) {
     return;
  }
}
```

Over:

```
void method() {
  if(!something) {
    // do stuff

  }
  return;

}

```

But, if a method is going to be a no-op, I generally prefer for it just not to be invoked. It's always a bit frustrating to read through code to realize that method calls are no-ops. Methods generally are called so that something can happen. For example, if a method no-ops because of null input, presumably the caller called it for something to happen, did the caller realize that they even provided a null input? IN such a case, I see something like:

```
void method(Sringg arg) {
  if(arg == null) {
     return; // << this is a silent failure, a BUG. Did the caller know they provided a null arg and were just doing a no-op?
  }
}
```